apis/src/providers/config.rs

```diff
@@ -0,0 +1,85 @@
+use std::collections::HashMap;
+use std::str::FromStr;
+
+use crate::common::{MoveConfirm, TileDesign, TileDots, TileRotation};
+
+use chrono::Utc;
+use codee::string::Base64;
+use codee::{binary::MsgpackSerdeCodec, string::FromToStringCodec};
+use leptos::*;
+use leptos_use::{use_cookie, use_cookie_with_options, SameSite, UseCookieOptions};
+use serde::{Deserialize, Serialize};
+use shared_types::GameSpeed;
+
+const USER_CONFIG_COOKIE: &str = "user_config";
+
+#[derive(Serialize, Deserialize, Clone)]
+pub struct ConfigOpts {
+    pub confirm_mode: HashMap<GameSpeed, MoveConfirm>,
+    pub tile_design: TileDesign,
+    pub tile_rotation: TileRotation,
+    pub tile_dots: TileDots,
+}
+#[derive(Clone, Serialize, Debug)]
+pub struct Config(pub Signal<ConfigOpts>);
+
+impl Config {
+    pub fn get_cookie() -> (Signal<Option<ConfigOpts>>, WriteSignal<Option<ConfigOpts>>) {
+        let exp = Utc::now()
+            .checked_add_signed(chrono::Duration::days(365))
+            .unwrap()
+            .timestamp();
+        use_cookie_with_options::<ConfigOpts, Base64<MsgpackSerdeCodec>>(
+            USER_CONFIG_COOKIE,
+            UseCookieOptions::default()
+                .same_site(SameSite::Lax)
```

Comment:
Why are we doing `Lax` here? Do we need the cookie to be cross-site?

Reply:
true, strict will suffice